R/teal_data_module.R

```diff
@@ -64,7 +64,16 @@ teal_data_module <- function(ui, server, label = "data module", once = TRUE) {
   checkmate::assert_function(ui, args = "id", nargs = 1)
   checkmate::assert_function(server, args = "id", nargs = 1)
   structure(
-    list(ui = ui, server = server),
+    list(
+      ui = ui,
+      server = function(id) {
+        data_out <- server(id)
+        decorate_err_msg(
+          prefix = "teal_data_module()",
+          assert_reactive(data_out, .var.name = "server")
+        )
```

Comment:
Hey @gogonzo. Copied from your message:

> applied decorate_err_msg so it indicates which function needs to be fixed (needs to return a reactive). We have access also to the label argument which can indicate which teal_data_module produced a failure. But I'm not sure how the decoration should look like. Maybe "teal_data_module(label = 'label...') or teal_data_module labeled as "label ..."

![image](https://github.com/user-attachments/assets/bb9a8b8d-4cef-4013-b199-451f8d920688)

I have a few comments on this:
- teal_data_module() needs to be code-formatted to become `teal_data_module()`. In checkmate it is done with `'` ticks
- just `teal_data_module()` might not be enough. This is quite tricky as this is not about this function inputs but actually what's returned from the module constructed with this function. 
- `server` is also misleading - this is about what server function returns. I think I would keep the current variable name so that devs can debug and rely on decoration for app/module developer.

All in all, I would propose the following: 
```suggestion
        decorate_err_msg(
          prefix = sprintf("From: 'teal_data_module()':\nA 'teal_data_module' with \"%s\" label:
", label),
          suffix = "Please make sure that this module returns a 'reactive` object containing 'teal_data' class of object.",
          assert_reactive(data_out)
        )
```

So that the final message would look like this:
```
Error:
From: 'teal_data_module()':
A 'teal_data_module' with "module label" label:
Assertion on 'data_out' failed: ...
Please make sure that this module returns a 'reactive` object containing 'teal_data' class of object.
```

I wanted to discuss this with you instead of pushing directly.

Other:
- we must introduce assert on `label` before using this so this must be added beforehand
- similar (same?) thing needs to be added to the `teal_transform_module()`

Reply:
@pawelru I agree with all above. One tiny thing, `prefix` and `suffix` is not precise as it refers to part of the word instead of part of the sentence. I suggest `pre`, `post`

Reply:
Pushed the changes. Please have a look. Closing this